### PR TITLE
xges: Effects and Markers Support

### PR DESCRIPTION
**Pull request by hwilkes-igalia**
Adds effects and markers support to the xges adapter.

GESEffects found below an xges clip translate to otio item effects.
GESMarkers found in an xges timeline translate to otio stack markers.

Also:
+ GstStructure was improved
+ GstCaps and GstCapsFeatures schemas were added
+ added tests
+ made small fixes and tidies

### CODE REVIEW

**Comment by thiblahute** [comment]
This looks pretty good to me, could we get a review?

@hwilkes-igalia It looks like this is still marked as WIP but afaik it is not a draft anymore, could you remove the flag?

File: contrib/opentimelineio_contrib/adapters/tests/tests_xges_adapter.py [hunk 1]
```diff
@@ -448,6 +565,40 @@ def name(name):
         return lambda inst, otio_item: inst.assertOtioAttrEqual(
             otio_item, "name", name)
 
+    @staticmethod
+    def effects(*effect_names):
+        """Return a test that the otio_item contains the effects"""
+        return lambda inst, otio_item: inst.assertOtioAttrPathEqualList(
```
[the remark this answers is not in the document]
Comment (hwilkes-igalia) [reply contrib/opentimelineio_contrib/adapters/tests/tests_xges_adapter.py]: it is used for calling `assertOtioAttrPathEqualList`

File: contrib/opentimelineio_contrib/adapters/tests/tests_xges_adapter.py [hunk 2]
```diff
@@ -89,12 +92,33 @@ def _make_clip(uri="file:///example", start=0, duration=1, name=""):
...
+    """Add a marker to an otio item"""
+    otio_item.markers.append(Marker(
+        name=name, color=color,
+        marked_range=_tm_range_from_secs(start, duration)))
+
+
+def _make_ges_marker(
+        position, otio_color=None, comment=None, metadatas=None):
+    """
+    Return a GESMarker with at the given timeline position (in seconds).
```
[the remark this answers is not in the document]
Comment (hwilkes-igalia) [reply contrib/opentimelineio_contrib/adapters/tests/tests_xges_adapter.py]: thanks. Changed it

**Comment by hwilkes-igalia** [comment]
> This looks pretty good to me, could we get a review?
> 
> @hwilkes-igalia It looks like this is still marked as WIP but afaik it is not a draft anymore, could you remove the flag?

Thanks for looking through.
I just changed the effect behaviour to write the GES `bin-description` into the metadata of an otio effect. When converting from otio to xges, we now use this `bin-description`. If it is not available, we give a warning and use the `effect_name` attribute of the otio effect as the `bin-description`.
Prior to this change, I just used the `effect_name` without a warning. But it is likely that an arbitrary `effect_name` would not work in GES as a `bin-description`.

In the future, if otio has standardised the effect names, this will need to be revised so we can properly convert `effect_name` into a bin description used in GES.

I'll remove the WIP now.

**Comment by tsaunier** [comment]
Thanks :-)

**Pull request merged**